bottlerocket/samples/Makefile.toml

```diff
@@ -0,0 +1,248 @@
+OVA_NAME = { value = "${PSEUDO_OVA_NAME}", condition = { env_not_set = ["OVA_NAME"] } }
+BOTTLEROCKET_AMI_ID = { value = "${PSEUDO_BOTTLEROCKET_AMI_ID}", condition = { env_not_set = ["BOTTLEROCKET_AMI_ID"] } }
+
+[tasks.instance-types]
+dependencies = ["set-env"]
+
+[tasks.instance-types.env]
+INSTANCE_TYPES = {script=['''
+if [ -z ${INSTANCE_TYPES} ]; then
+    if [ ${ARCHITECTURE} = "x86_64" ]; then
+        if [ ${GPU} = "true" ]; then
+            echo "[\"g4dn.xlarge\"]"
+        else
+            echo "[\"m5.large\"]"
+        fi
+    else
+        if [ ${GPU} = "true" ]; then
+            echo "[\"g5g.xlarge\"]"
+        else
+            echo "[\"m6g.large\"]"
+        fi
+    fi
+else
+    echo ${INSTANCE_TYPES}
+fi
+''']}
+
+[tasks.create-inner]
+dependencies = ["set-env", "instance-types"]
+script_runner = "bash"
+script = '''
+eval "cat > ${OUTPUT_FILE} << EOF
+$(< ${CLUSTER_TYPE}/${TEST_TYPE}.yaml)
+EOF
+" 2> /dev/null
+
+echo "${OUTPUT_FILE}"
+'''
+
+[tasks.create-inner.env]
+OUTPUT_FILE = {script=['''
+if [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "migration" ]; then
+    echo "output/${CLUSTER_NAME}-migration.yaml"
+elif [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "workload" ]; then
+    echo "output/${CLUSTER_NAME}-workload.yaml"
+else
+    echo "output/${CLUSTER_NAME}.yaml"
+fi
+''']}
+
+[tasks.create]
+script = '''
+#!@duckscript
+set_env TEST_TYPE ${1}
+cm_run_task ${1}
+cm_run_task create-inner
+'''
+
+[tasks.install-controller]
+script = "cli install --controller-uri ${CONTROLLER_IMAGE_URI}"
+
+[tasks.add-secrets]
+dependencies = ["install-controller"]
+script = '''
+if [ $(echo ${TEST_TYPE} | tr '-' ' ' | awk '{print $1}'") = "vmware" ]; then
+    cli add-secret map  \
+     --name "vsphere-creds" \
+     "username=${GOVC_USERNAME}" \
+     "password=${GOVC_PASSWORD}"
+elif [ ${CLUSTER_TYPE} = "kind" ]; then
+    ACCESS_KEY_ID=$(aws configure get aws_access_key_id)
+    SECRET_ACCESS_KEY=$(aws configure get aws_secret_access_key)
+    cli add-secret map  \
+     --name "aws-creds" \
+     "ACCESS_KEY_ID=${ACCESS_KEY_ID}" \
+     "SECRET_ACCESS_KEY=${SECRET_ACCESS_KEY}"
+fi
+'''
+
+[tasks.run-inner]
+dependencies = ["install-controller", "add-secrets"]
+script = "cli run file ${OUTPUT_FILE}"
+
+[tasks.run]
+script = '''
+#!@duckscript
+set_env TEST_TYPE ${1}
+cm_run_task ${1}
+cm_run_task create-inner
+cm_run_task run-inner
+'''
+
+[tasks.ecs-migration-test.env]
+VARIANT = "aws-ecs-1"
+
+[tasks.ecs-test.env]
+VARIANT = "aws-ecs-1"
+
+[tasks.ecs-workload-test.env]
+VARIANT = "aws-ecs-1-nvidia"
+
+[tasks.sonobuoy-migration-test.env]
+VARIANT = "aws-k8s-1.24"
+
+[tasks.sonobuoy-test.env]
+VARIANT = "aws-k8s-1.24"
+
+[tasks.k8s-workload-test.env]
+VARIANT = "aws-k8s-1.24-nvidia"
+
+[tasks.vmware-migration-test.env]
+VARIANT = "vmware-k8s-1.24"
+
+[tasks.vmware-sonobuoy-test.env]
+VARIANT = "vmware-k8s-1.24"
+
+[tasks.get-samples.env]
+SAMPLES = {script=["echo $(ls ${CLUSTER_TYPE})"]}
```

Comment:
```suggestion
SAMPLES = {script=["ls ${CLUSTER_TYPE}"]}
```
Does this work?

Reply:
That only gets the last filename for some reason. Using `ls -xm` to put commas in between the files only stores the last 2 in the variable. This is the most reliable way I've found that works.